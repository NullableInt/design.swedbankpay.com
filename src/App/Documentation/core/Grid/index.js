import React from "react";

import ComponentPreview from "../../utils/ComponentPreview";
import DocToc from "../../utils/DocToc";

const HowItWorks = () => (
    <>
        <h2 id="how-it-works">How it works</h2>
        <p>The grid system is heavily based on Bootstrap{"'"}s grid from v.4.</p>
        <ComponentPreview language="html" showCasePanel={true} codeFigure={true} removeOuterTag={true}>
            <div className="showcase-grid">
                <div className="container">
                    <div className="row">
                        <div className="col-sm">
                            One of three columns
                        </div>
                        <div className="col-sm">
                            One of three columns
                        </div>
                        <div className="col-sm">
                            One of three columns
                        </div>
                    </div>
                </div>
            </div>
        </ComponentPreview>
    </>
);

const AutoLayoutColumns = () => (
    <>
        <h2 id="auto-layout-columns">Auto-layout columns</h2>
        <p>Utilize breakpoint-specific column classes for easy column sizing without an explicit numbered class like <code className="token property">.col-sm-6</code>.</p>
        <ComponentPreview language="html" showCasePanel={true} codeFigure={true} removeOuterTag={true}>
            <div className="showcase-grid">
                <div className="container">
                    <div className="row">
                        <div className="col">
                            1 of 2
                        </div>
                        <div className="col">
                            2 of 2
                        </div>
                    </div>
                    <div className="row">
                        <div className="col">
                            1 of 3
                        </div>
                        <div className="col">
                            2 of 3
                        </div>
                        <div className="col">
                            3 of 3
                        </div>
                    </div>
                </div>
            </div>
        </ComponentPreview>
    </>
);

const GridText = () => (
    <div className="col-md-12 col-lg-10 doc-body">
        <p className="lead">Use our grid system...</p>
        <HowItWorks />
        <AutoLayoutColumns />
    </div>
);

const Grid = () => {
    return (
        <div className="doc-container">
            <div className="row">
                <GridText />
                <DocToc component={GridText} />
            </div>
        </div>
    );
};

export default Grid;
