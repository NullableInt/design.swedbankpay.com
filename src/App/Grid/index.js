import React, { Component } from "react";

class Examples extends Component {
    render () {
        return (
            <>
                <h1>GRID</h1>
                <div className="container">
                    <div className="row">
                        <div className="col-md-1 offset-md-2"></div>
                        <div className="col-md-2"></div>
                        <div className="col-md-3"></div>
                        <div className="col-md-4"></div>
                        <div className="col-md-5"></div>
                        <div className="col-md-6"></div>
                        <div className="col-md"></div>
                        <div className="col-md-auto"></div>
                    </div>
                </div>
            </>
        );
    }
}

export default Examples;
