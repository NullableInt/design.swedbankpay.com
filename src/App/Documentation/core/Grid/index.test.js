import React from "react";
import { shallow } from "enzyme";

import Grid, { HowItWorks, AutoLayoutColumns, GridText } from "./index";

describe("Core: Grid", () => {
    it("is defined", () => {
        expect(Grid).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(
            <Grid />
        );

        expect(wrapper).toMatchSnapshot();
    });

    describe("HowItWorks", () => {
        it("is defined", () => {
            expect(HowItWorks).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <HowItWorks />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("AutoLayoutColumns", () => {
        it("is defined", () => {
            expect(AutoLayoutColumns).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <AutoLayoutColumns />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("GridText", () => {
        it("is defined", () => {
            expect(GridText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <GridText />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
