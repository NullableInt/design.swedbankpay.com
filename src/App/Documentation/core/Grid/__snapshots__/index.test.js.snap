// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core: Grid AutoLayoutColumns renders 1`] = `
<React.Fragment>
  <h2
    id="auto-layout-columns"
  >
    Auto-layout columns
  </h2>
  <p>
    Utilize breakpoint-specific column classes for easy column sizing without an explicit numbered class like 
    <code
      className="token property"
    >
      .col-sm-6
    </code>
    .
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <div
      className="showcase-grid"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            1 of 2
          </div>
          <div
            className="col"
          >
            2 of 2
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            1 of 3
          </div>
          <div
            className="col"
          >
            2 of 3
          </div>
          <div
            className="col"
          >
            3 of 3
          </div>
        </div>
      </div>
    </div>
  </ComponentPreview>
</React.Fragment>
`;

exports[`Core: Grid GridText renders 1`] = `
<div
  className="col-md-12 col-lg-10 doc-body"
>
  <p
    className="lead"
  >
    Use our grid system...
  </p>
  <HowItWorks />
  <GridOptions />
  <AutoLayoutColumns />
</div>
`;

exports[`Core: Grid HowItWorks renders 1`] = `
<React.Fragment>
  <h2
    id="how-it-works"
  >
    How it works
  </h2>
  <p>
    The grid system is heavily based on Bootstrap
    '
    s grid from v.4.
  </p>
  <p>
    Most classes from Bootstrap is available. Click 
    <a
      href="http://getbootstrap.com/docs/4.1/layout/grid/"
    >
      here
    </a>
     to read more about it.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <div
      className="showcase-grid"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-sm"
          >
            One of three columns
          </div>
          <div
            className="col-sm"
          >
            One of three columns
          </div>
          <div
            className="col-sm"
          >
            One of three columns
          </div>
        </div>
      </div>
    </div>
  </ComponentPreview>
  <p>
    The above example creates three equal-width columns on small, medium, large, extra large, and extra extra large devices using our predefined grid classes. Those columns are centered in the page with the parent 
    <code
      className="token property"
    >
      .container
    </code>
    .
  </p>
</React.Fragment>
`;

exports[`Core: Grid renders 1`] = `
<div
  className="doc-container"
>
  <div
    className="row"
  >
    <GridText />
    <DocToc
      component={[Function]}
    />
  </div>
</div>
`;
